packages/grafana-runtime/src/utils/DataSourceWithBackend.test.ts

```diff
@@ -0,0 +1,83 @@
+import { BackendSrv } from 'src/services';
+import { DataSourceWithBackend } from './DataSourceWithBackend';
+import { DataSourceJsonData, DataQuery, DataSourceInstanceSettings, DataQueryRequest } from '@grafana/data';
+
+class MyDataSource extends DataSourceWithBackend<DataQuery, DataSourceJsonData> {
+  constructor(instanceSettings: DataSourceInstanceSettings<DataSourceJsonData>) {
+    super(instanceSettings);
+  }
+}
+
+const mockDatasourceRequest = jest.fn();
+
+const backendSrv = ({
+  datasourceRequest: mockDatasourceRequest,
+} as unknown) as BackendSrv;
+
+jest.mock('../services', () => ({
+  getBackendSrv: () => backendSrv,
+}));
+jest.mock('..', () => ({
+  config: {
+    bootData: {
+      user: {
+        orgId: 77,
+      },
+    },
+    datasources: {
+      sample: {
+        id: 8674,
+      },
+    },
+  },
+}));
+
+describe('DataSourceWithBackend', () => {
+  test('check the executed queries', () => {
+    const settings = {
+      name: 'test',
+      id: 1234,
+      jsonData: {},
+    } as DataSourceInstanceSettings<DataSourceJsonData>;
+
+    mockDatasourceRequest.mockReset();
+    mockDatasourceRequest.mockReturnValue(Promise.resolve({}));
+    const ds = new MyDataSource(settings);
+    ds.query({
+      maxDataPoints: 10,
+      intervalMs: 5000,
+      targets: [{ refId: 'A' }, { refId: 'B', datasource: 'sample' }],
+    } as DataQueryRequest);
+
+    const mock = mockDatasourceRequest.mock;
+    expect(mock.calls.length).toBe(1);
+
+    const args = mock.calls[0][0];
+    expect(args).toMatchInlineSnapshot(`
+      Object {
+        "data": Object {
+          "queries": Array [
+            Object {
+              "datasourceId": 1234,
+              "intervalMs": 5000,
+              "maxDataPoints": 10,
+              "orgId": 77,
+              "refId": "A",
+            },
+            Object {
+              "datasource": "sample",
+              "datasourceId": 8674,
+              "intervalMs": 5000,
+              "maxDataPoints": 10,
+              "orgId": 77,
+              "refId": "B",
+            },
+          ],
+        },
+        "method": "POST",
+        "requestId": undefined,
+        "url": "/api/ds/query",
+      }
+    `);
```

Comment:
nit: why not use toEqual?
```suggestion
    expect(args).toEqual(
      {
        data: {
          queries: [
            {
              datasourceId: 1234,
              intervalMs: 5000,
              maxDataPoints: 10,
              orgId: 77,
              refId: 'A',
            },
            {
              datasource: 'sample',
              datasourceId: 8674,
              intervalMs: 5000,
              maxDataPoints: 10,
              orgId: 77,
              refId: 'B',
            },
          ],
        },
        method: 'POST',
        requestId: undefined,
        url: '/api/ds/query',
      }
    `);
```

Reply:
`toMatchInlineSnapshot` will fill in the snapshot for you :)  toEqual... you have to define it explicilty.

It seems like 6 vs 1/2 dozen... or is there a reason to use toEqual?

Reply:
I don't have any strong opinions about this, just curious what use case was for toMatchInlineSnapshot :)

Reply:
Huh I did not know there was such thing in jest. Interesting thing. Otherwise I would say also say it is easier to work with js objects than with plain string but this being generated that does not matter.